mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -1353,6 +1371,18 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 		ResourceLabels: expandStringMap(d, "resource_labels"),
 	}
 
+	if v, ok := d.GetOk("enable_autopilot"); ok {
+		cluster.Autopilot = &containerBeta.Autopilot{
+			Enabled:         v.(bool),
+			ForceSendFields: []string{"Enabled"},
+		}
+		if v.(bool) == true {
+			// Set preconfigured AutoGKE defaults
+			cluster.ShieldedNodes.Enabled = true
```

Comment:
`enable_shielded_nodes` has a default of `false` and it conflicts with this field. How don't we get a permadiff? Should we ensure that these values are set correctly at plan time using a `CustomizeDiff` instead?

Reply:
You are totally correct, I just missed this. Adding a `CustomizeDiff` to check these values is possible, but is there anything we can do to prevent users from having to set the fields at all? My thoughts are making the two fields Optional+Computed and letting the API handle it defaults.

Reply:
I have a minor preference for the `CustomizeDiff` & having configs be explicit about what's configured, but not a strong one.

Reply:
Looking more into the autopilot use case, I'm seeing other previously non-computed fields (`ip_allocation_policy` and `workload_identity_config`) being set by the API which I believe forces us to use `Computed` for these. I think O+C for the preconfigured defaults gives a better user experience, so I want to try that out. I don't think it will disrupt the standard use cases, but I'll be sure to test more thoroughly this time.

Reply:
@rileykarson Seeing as this needs more work, feel free to unfollow and I'll re-request when it is ready. I'll be pushing commits to this PR to take advantage of TC tests.